pkg/ui/src/views/queries/statementDetails.tsx

```diff
@@ -0,0 +1,209 @@
+import { SummaryBar, SummaryHeadlineStat } from "src/views/shared/components/summaryBar";
+import * as protos from "src/js/protos";
+
+import { countBarChart, latencyBarChart } from "./barCharts";
+
+type StatementStatistics = protos.cockroach.sql.CollectedStatementStatistics$Properties;
+
+interface StatementDetailsOwnProps {
+  statement: StatementStatistics;
+  refreshQueries: typeof refreshQueries;
+}
+
+type StatementDetailsProps = StatementDetailsOwnProps & RouterState;
+
+interface NumericStat {
+  mean?: number;
+  squared_diffs?: number;
+}
+
+function variance(stat: NumericStat, count: number) {
+  return stat.squared_diffs / (count - 1);
+}
+
+interface NumericStatRow {
+  name: string;
+  value: NumericStat;
+}
+
+interface NumericStatTableProps {
+  rows: NumericStatRow[];
+  count: number;
+  format: (v: number) => string;
+}
+
+class NumericStatTable extends React.Component<NumericStatTableProps> {
+  static defaultProps = {
+    format: (v: number) => `${v}`,
+  };
+
+  render() {
+    return (
+      <table className="numeric-stats-table">
+        <thead>
+          <tr className="numeric-stats-table__row--header">
+            <th className="numeric-stats-table__cell" />
+            <th className="numeric-stats-table__cell">Mean</th>
+            <th className="numeric-stats-table__cell">Variance</th>
+          </tr>
+        </thead>
+        <tbody style={{ textAlign: "right" }}>
+          {
+            this.props.rows.map((row: NumericStatRow) => {
+              return (
+                <tr className="numeric-stats-table__row--body">
+                  <th className="numeric-stats-table__cell" style={{ textAlign: "left" }}>{ row.name }</th>
+                  <td className="numeric-stats-table__cell">{ this.props.format(row.value.mean) }</td>
+                  <td className="numeric-stats-table__cell">{ this.props.format(variance(row.value, this.props.count)) }</td>
+                </tr>
+              );
+            })
+          }
+        </tbody>
+      </table>
+    );
+  }
+}
+
+class StatementDetails extends React.Component<StatementDetailsProps> {
+  componentWillMount() {
+    this.props.refreshQueries();
+  }
+
+  componentWillReceiveProps() {
+    this.props.refreshQueries();
+  }
+
+  render() {
+    if (!this.props.statement) {
+      return "loading...";
+    }
+
+    const { stats } = this.props.statement;
+
+    const count = FixLong(stats.count).toInt();
+    const firstAttemptCount = FixLong(stats.first_attempt_count).toInt();
+
+    const countBar = countBarChart();
+    const latencyBar = latencyBarChart();
+
+    return (
+      <div>
+        <Helmet>
+          <title>{`Details | Statements`}</title>
+        </Helmet>
+        <section className="section">
+          <h2>Statement Details</h2>
+          <div className="content l-columns">
+            <div className="l-columns__left">
+              <SqlBox value={ this.props.statement.key.query } />
```

Comment:
design tweak: until we figure out how to use @mjibson's sqlfmt to pretty print queries, I suggest applying this diff to `SqlBox` to just wrap the query text:

```diff
diff --git a/pkg/ui/src/views/shared/components/sql/sqlhighlight.styl b/pkg/ui/src/views/shared/components/sql/sqlhighlight.styl
index 3daaf9fdf..b3d15397b 100644
--- a/pkg/ui/src/views/shared/components/sql/sqlhighlight.styl
+++ b/pkg/ui/src/views/shared/components/sql/sqlhighlight.styl
@@ -5,6 +5,7 @@
   padding 20px
   width 100%
   margin-bottom 10px
+  white-space pre-wrap
 
 .sql-highlight.hljs
     background-color $headings-color
```

Reply:
Done.